lib/lfrfid/protocols/protocol_nexwatch.c

```diff
@@ -0,0 +1,325 @@
+    memset(protocol->negative_encoded_data, 0, NEXWATCH_ENCODED_DATA_SIZE);
+    memset(protocol->corrupted_encoded_data, 0, NEXWATCH_ENCODED_DATA_SIZE);
+    memset(protocol->corrupted_negative_encoded_data, 0, NEXWATCH_ENCODED_DATA_SIZE);
+};
+
+static bool protocol_nexwatch_check_preamble(uint8_t* data, size_t bit_index) {
+    // 01010110
+    if(bit_lib_get_bits(data, bit_index, 8) != 0b01010110) return false;
+    return true;
+}
+
+static uint8_t protocol_nexwatch_parity_swap(uint8_t parity) {
+    uint8_t a = (((parity >> 3) & 1));
+    a |= (((parity >> 1) & 1) << 1);
+    a |= (((parity >> 2) & 1) << 2);
+    a |= ((parity & 1) << 3);
+    return a;
+}
+
+static uint8_t protocol_nexwatch_parity(const uint8_t hexid[5]) {
+    uint8_t p = 0;
+    for(uint8_t i = 0; i < 5; i++) {
+        p ^= ((hexid[i]) & 0xF0) >> 4;
+        p ^= ((hexid[i]) & 0x0F);
+    }
+    return protocol_nexwatch_parity_swap(p);
+}
+
+static uint8_t protocol_nexwatch_checksum(uint8_t magic, uint32_t id, uint8_t parity) {
+    uint8_t a = ((id >> 24) & 0xFF);
+    a -= ((id >> 16) & 0xFF);
+    a -= ((id >> 8) & 0xFF);
+    a -= (id & 0xFF);
+    a -= magic;
+    a -= (bit_lib_reverse_8_fast(parity) >> 4);
+    return bit_lib_reverse_8_fast(a);
+}
+
+static bool protocol_nexwatch_can_be_decoded(uint8_t* data) {
+    if(!protocol_nexwatch_check_preamble(data, 0)) return false;
+
+    // Check for reserved word (32-bit)
+    if(bit_lib_get_bits_32(data, 8, 32) != 0) {
+        return false;
+    }
+
+    uint8_t parity = bit_lib_get_bits(data, 76, 4);
+
+    // parity check
+    // from 32b hex id, 4b mode
+    uint8_t hex[5] = {0};
+    for(uint8_t i = 0; i < 5; i++) {
+        hex[i] = bit_lib_get_bits(data, 40 + (i * 8), 8);
+    }
+    //mode is only 4 bits.
+    hex[4] &= 0xf0;
+    uint8_t calc_parity = protocol_nexwatch_parity(hex);
+
+    if(calc_parity != parity) {
+        return false;
+    }
+
+    return true;
+}
+
+static bool protocol_nexwatch_decoder_feed_internal(bool polarity, uint32_t time, uint8_t* data) {
+    time += (NEXWATCH_US_PER_BIT / 2);
+
+    size_t bit_count = (time / NEXWATCH_US_PER_BIT);
+    bool result = false;
+
+    if(bit_count < NEXWATCH_ENCODED_BIT_SIZE) {
+        for(size_t i = 0; i < bit_count; i++) {
+            bit_lib_push_bit(data, NEXWATCH_ENCODED_DATA_SIZE, polarity);
+            if(protocol_nexwatch_can_be_decoded(data)) {
+                result = true;
+                break;
+            }
+        }
+    }
+
+    return result;
+}
+
+static void protocol_nexwatch_descramble(uint32_t* id, uint32_t* scrambled) {
+    // 255 = Not used/Unknown other values are the bit offset in the ID/FC values
+    const uint8_t hex_2_id[] = {31, 27, 23, 19, 15, 11, 7, 3, 30, 26, 22, 18, 14, 10, 6, 2,
+                                29, 25, 21, 17, 13, 9,  5, 1, 28, 24, 20, 16, 12, 8,  4, 0};
+
+    *id = 0;
+    for(uint8_t idx = 0; idx < 32; idx++) {
+        if(hex_2_id[idx] == 255) continue;
```

Comment:
this expression is always false

Reply:
What I found:

<img width="731" alt="Screenshot 2023-05-26 at 20 00 14" src="https://github.com/flipperdevices/flipperzero-firmware/assets/10697207/84766235-b856-40dd-ad01-3e29950d9f9e">

-----

Issue comes from proxmark code

It has same check thats always false 

I think if it works with current `hex_2_id` table, this `if(hex_2_id[idx] == 255) continue;` can be safely removed without any issues

Reply:
yes please, remove it and open issue in PM repository

Reply:
Done